content/scalability/docs/workloads.md

```diff
@@ -0,0 +1,89 @@
+
+For example, the default ALB targets is 1000. If you have a service with more than 1000 endpoints you will need to increase the quota or split the service across multiple ALBs or use Kubernetes Ingress. The default NLB targets is 3000, but is limited to 500 targets per AZ. If your cluster runs more than 500 pods for an NLB service you will need to use more AZs or request a quota increase.
+
+Kubernetes ingress allows you to expose multiple Kubernetes services from a single load balancer by running a proxy inside your cluster. You can read more about ingress options in the [Kubernetes documentation](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/).
+
+## Use EndpointSlices instead of Endpoints
+
+When discovering pods that match a service label you should use [EndpointSlices](https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/) instead of Endpoints. Endpoints were a simple way to expose services at small scales but large services that automatically scale or have updates causes a lot of traffic on the Kubernetes control plane. EndpointSlices have automatic grouping which enable things like topology aware hints.
+
+Not all controllers use EndpointSlices by default. You should verify your controller settings and enable it if needed. For the [AWS Load Balancer Controller](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/deploy/configurations/#controller-command-line-flags) you should enable the --enable-endpoint-slices optional flag to use EndpointSlices.
+
+## Use immutable and external secrets if possible
+
+The kubelet keeps a cache of the current keys and values for the Secrets that are used in volumes for pods on that node. The kubelet sets a watch on the Secrets to detect changes. As the cluster scales, the growing number of watches can negatively impact the API server performance.
+
+There are two strategies to reduce the number of watches on Secrets:
+
+* For applications that don’t need access to Kubernetes resources, you can disable auto-mounting service account secrets by setting automountServiceAccountToken: false
+* If your application’s secrets are static and will not be modified in the future, mark the [Secret as Immutable](https://kubernetes.io/docs/concepts/configuration/secret/#secret-immutable). The kubelet does not need maintain a watch for immutable secrets.
+
+To disable automatically mounting a service account to pods you can use the following setting in your workload. You can override these settings if specific workloads need a service account.
+
+```
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: app
+automountServiceAccountToken: true
+```
+
+Monitor the number of secrets in the cluster before it exceeds the limit of 10,000. You can get a total count of secrets in a cluster with the following commend.
+
+```
+kubectl get secrets -A | wc -l
```

Comment:
won't this issue a list against the API? lists=bad. Maybe we can get this from kube-state-metrics instead or through a CW Log Insights Query.